tuf/api/metadata.py

```diff
@@ -1258,17 +1267,173 @@ def to_dict(self) -> Dict[str, Any]:
+        """Returns the dict representation of self."""
+        return {
+            "delegation_hash_prefix_len": self.hash_prefix_len,
+            "bin_name_prefix": self.bin_name_prefix,
+            **self.unrecognized_fields,
+        }
+
+    def get_bin_name(self, bin_number: int = 0) -> str:
+        """Return the name of bin with number ``bin_number``.
+        The ``bin_number`` default is 0 as that is used to uniquely identify
+        delegated roles containing succinct hash delegations from the rest.
+
+        Args:
+            bin_number: Number of the bin that we want the name for.
+        """
+        # Add zero padding if necessary.
+        suffix = f"{bin_number:0{self.zero_padding}}"
+        return f"{self.bin_name_prefix}-{suffix}"
+
+    def get_all_bin_names(self) -> Iterator[str]:
+        """Yield the names of all different bins one by one."""
+        for i in range(0, self.number_of_bins):
+            yield self.get_bin_name(i)
+
+    def is_bin(self, role_name: str) -> bool:
+        """Determines whether the given ``role_name`` is in one of
+        the bins that ``SuccinctHashDelegations`` is trusted to provide.
+
+        Args:
+            role_name: The name of the delegated role to check against.
+        """
+        desired_prefix = self.bin_name_prefix + "-"
+
+        if role_name.startswith(desired_prefix):
+            suffix = role_name[len(desired_prefix) :]
+            try:
+                num = int(suffix)
+                if 0 <= num < self.number_of_bins:
+                    return True
+            except ValueError:
+                # suffix is not a number
+                return False
+
+        return False
+
+    def _find_bin_for_bits(self, hash_bits_representation: str) -> str:
+        """Helper function for find_delegation calculating the actual rolename.
+
+        Args:
+            hash_bits_representation: binary bit representation of the target
+                hash.
+        """
+
+        bit_length = self.hash_prefix_len
+        # Get the first bit_length of bits and then cast them to decimal.
+        bin_number = int(hash_bits_representation[:bit_length], 2)
+
+        return self.get_bin_name(bin_number)
+
+    def find_bin(self, target_filepath: str) -> str:
+        """Calculates the name of the bin responsible for ``target_filepath``.
+
+        Args:
+            target_filepath: URL path to a target file, relative to a base
+                targets URL.
+        """
+        hasher = sslib_hash.digest(algorithm="sha256")
+        hasher.update(target_filepath.encode("utf-8"))
+        # Get the bit representation of the hash. The bit representation is
+        # calculated for a small number of symbols required by hash_prefix_len.
+        needed_bits = self.hash_prefix_len  # aliased for emphasis
+        needed_bytes = math.ceil(needed_bits / 8)
+        hash_bits = "".join(
+            f"{one_byte:08b}" for one_byte in hasher.digest()[:needed_bytes]
```

Comment:
i'm not sure if this is a bug or just bad naming but a hexadecimal digit is not equal to a byte (but see my other comment: I'm proposing removing this code altogether)

Reply:
oh I may be misunderstanding the hashlib use here -- digest() maybe actually returns the bytes and not hex: in that case never mind me, I'm the one making a mistake